src/test/java/org/zalando/problem/spring/web/advice/http/NotAcceptableAdviceTraitTest.java

```diff
@@ -30,7 +30,7 @@ void notAcceptable() throws Exception {
     @Test
     void notAcceptableNoProblem() throws Exception {
         mvc().perform(request(GET, "http://localhost/api/handler-ok")
-                .accept("application/atom+xml"))
+                .accept("image/png"))
                 .andExpect(status().isNotAcceptable())
                 .andExpect(header().doesNotExist("Content-Type"));

```

Comment:
Shouldn't the `Content-Type` be `MediaTypes.PROBLEM` ?

Reply:
Yes. I just haven't figured out a way yet to force Spring to do the right thing there. The content negotiation is done up front and my code never runs...

Reply:
You can ignore the Accept header in the Spring Configuration using something like this:
```
public void configureContentNegotiation(ContentNegotiationConfigurer configurer) {

	configurer.favorPathExtension(false);
	configurer.ignoreAcceptHeader(true);
	configurer.defaultContentType(MediaType.APPLICATION_JSON);

}
```
You could try adding this to your test configuration.